pkg/collector/python/py3_checker.go

```diff
@@ -31,7 +33,7 @@ func validatePython3(moduleName string, modulePath string) ([]string, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), linterTimeout)
 	defer cancel()
 
-	cmd := exec.CommandContext(ctx, pythonBinPath, "-m", "a7", modulePath)
+	cmd := exec.CommandContext(ctx, pythonBinPath, "-m", "pylint", "-f", "json", "--py3k", "-d", "W1618", "--persistent", "no", "--exit-zero", modulePath)
```

Comment:
I believe that with this approach, we do not try to leave out `site-packages` as we did in `a7`: https://github.com/DataDog/datadog-checks-shared/blob/62c5c4ca2e42a62ef9866323f5697fbb6ee6b508/a7/validate.py#L35-L39, which might lead to slower execution. Are we fine with this?

Reply:
Yes, now that we removed the timeout, it should be fine.

Reply:
2 follow-up questions actually:

1. does it only make the execution slower, or does it also have an impact on the resource (CPU/mem/IO) usage on the host? Do you have data on the use cases that forced us to leave out `site-packages` in the first place?
2. although we run the validation concurrently (so the validation shouldn't block the main collector goroutine anymore), we do hold a lock here: https://github.com/DataDog/datadog-agent/blob/e3de3896afbbaf831af5205fffc3e827a4ed3db9/pkg/collector/python/loader.go#L246-L247 , it may have an impact on functionality of the Agent while this runs. If this is the only potential side effect I can look into what we can do to avoid holding the lock for too long

Reply:
1. It does have resource impact, as it's loading python module. The use cases we found when removing `site-packages` was the parsing of client libraries like kubernetes was taking a while, and pylint checks dependencies for some dumb reasons.

2. I don't know about that unfortunately.

Reply:
cc @xlucas: do you recall who found out about the impact of the python3 linter on custom checks that import `kubernetes`?

Reply:
did some testing on the custom check that we use internally and that imports `kubernetes` (called `zk_homogeneous_config`): using `pylint` directly instead of `a7` (which in practical terms equals not "cleaning up" `site-packages`) makes the command use ~100MB of memory instead of ~20MB, and makes it complete in 50sec instead of 4sec.

So here's the potential impact I can think of:
1. increase in memory usage of the agent when loading checks (so generally at startup). Could potentially make the Agent container get OOM-killed in extreme cases.
2. status page and agent expvars unavailable while the validation command runs (because of the lock I mentioned in previous comment, held here: https://github.com/DataDog/datadog-agent/blob/e3de3896afbbaf831af5205fffc3e827a4ed3db9/pkg/collector/python/loader.go#L246-L247)

I'll try to get a PR ready today to mitigate the effects of `2.` caused by the lock.

Still, it'll be interesting to track the impact of this at Agent startup when we QA 6.15.0 RCs on our k8s clusters, cc @CharlyF